src/test/java/org/cactoos/http/HtRetryWireTest.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.cactoos.http;
+
+import org.cactoos.io.InputOf;
+import org.cactoos.text.TextOf;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+/**
+ * Test case for {@link HtRetryWire}.
+ *
+ * @author Victor Noel (victor.noel@crazydwarves.org)
+ * @version $Id$
+ * @since 0.1
+ * @checkstyle JavadocMethodCheck (500 lines)
+ * @checkstyle JavadocVariableCheck (500 lines)
+ */
+public final class HtRetryWireTest {
+
+    @Rule
+    public final ExpectedException expected = ExpectedException.none();
+
+    private int retries = 1;
+
+    @Test
+    public void runsHtRetryWireMultipleTimes() throws Exception {
+        final String txt = "out";
+        final int max = 3;
+        MatcherAssert.assertThat(
+            new TextOf(
+                new HtRetryWire(
+                    input -> {
+                        if (this.retries < max) {
+                            this.retries = this.retries + 1;
+                            throw new IllegalArgumentException("Won't happen");
+                        }
+                        return new InputOf(txt);
+                    },
+                    max
+                ).send(new InputOf("in"))
+            ).asString(),
+            Matchers.equalTo(txt)
+        );
+    }
+
+    @Test
+    public void failsAfterMaxRetries() throws Exception {
+        this.expected.expect(IllegalArgumentException.class);
```

Comment:
@victornoel add also `expectMessage`, so we will be sure it was this `IllegalArgumentException`

Reply:
@krzyk Ok, I've used a `String` in this case and not a `Text` to represent the message because of the overhead it would incur to call `asString()` on it everytime it is used, but I'm open to changing that.

Reply:
(@krzyk by overhead I mean mental writing/reading overhead, not runtime overhead of course)